### PR TITLE
Issue 2/safety controller node

### PR DESCRIPTION
**Pull request by pbaughman**
Resolves issue #2
We're providing the safety back-up, since it's a bit complicated.

### Changes
 - Adding infrastructure for ROS tests
 - Added safety controller executable, launch file, and code
 - Added ROS test for safety controller node

### How to test:
checked on kuri-0001c.
```
>sudo service gizmo stop
set up workspace, source devel setup.sh . . .
roslaunch mobile_base_driver kuri_drive.launch
roslaunch kuri_edu safety_controller.launch
```
Touch the bumper and observe that the robot backs up
rostopic echo /mobile_base/safety_status
Observe that safety statuses are resolved automatically

### CODE REVIEW

File: kuri_edu/src/kuri_edu/safety_controller.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    def run(self):
+
+        # Indicate to the world that we're running and ready to go:
+        self._pub.publish(
+            mayfield_msgs.msg.NodeStatus("safety_controller", True)
+        )
+
+        rate = rospy.Rate(self.SAFETY_HZ)  # 10hz by default
+        while not rospy.is_shutdown():
```
Comment (pbaughman) [inline kuri_edu/src/kuri_edu/safety_controller.py:68]: Here's where the magic happens:

File: kuri_edu/src/kuri_edu/safety_controller.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            try:
+
+                current_status = self._safety_client.get_safety_status()
+
+                if self.HANDLED_EVENTS.intersection(current_status) != set():
```
Comment (spmaniato) [inline kuri_edu/src/kuri_edu/safety_controller.py]: Minor: I think you can omit `!= set()`. The empty `set` will evaluate to `False` I believe.

```python
>>> if not set():
...     print "do stuff"
do stuff
```

File: kuri_edu/src/kuri_edu/safety_controller.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    with self._sync_lock:
+                        self._block_twists = True
+
+                    # Override the safety status and back the robot up
```
Comment (spmaniato) [inline kuri_edu/src/kuri_edu/safety_controller.py]: Was this block of code also meant to be within the lock's scope?

**Review by spmaniato — COMMENTED** [review]
LGTM. Two inline comments

File: kuri_edu/src/kuri_edu/safety_controller.py (hunk 3 above)
Comment (pbaughman) [reply kuri_edu/src/kuri_edu/safety_controller.py]: No, but I'll add a comment explaining why

**Review by spmaniato — APPROVED** [review]

**Pull request merged**